node/lib/util/commit.js

```diff
@@ -1271,3 +1335,181 @@ exports.getCommitStatus = co.wrap(function *(repo, cwd, options) {
+
+    subNames.forEach(subName => {
+        const sub = submodules[subName];
+
+        const workdir = sub.workdir;
+        if (null === workdir) {
+            // Cannot generate a commit for a closed submodule.
+
+            return;                                                   // RETURN
+        }
+        const subStat = workdir.status;
+        if (subStat.isIndexClean()) {
+            // No need to do anything if the submodule has no staged files.
+
+            return;                                                   // RETURN
+        }
+
+        result += `\
+# -----------------------------------------------------------------------------
+
+# <${subName}> enter message for '${subName}' above this line; delete this \
+line to skip committing '${subName}'
+`;
+
+        result += exports.prefixWithPound(exports.formatStatus(subStat, ""));
+    });
+
+    result += `\
+#
+# Please enter the commit message(s) for your changes.  The message for a
+# repo will be composed of all lines not beginning with '#' that come before
+# its tag, but after any other tag (or the beginning of the file).  Tags are
+# lines beginning with '# <sub-repo-name>', or '# <*>' for the meta-repo.
+# If the tag for a repo is removed, no commit will be generated for that repo.
+# If you do not provide a commit message for a sub-repo, the commit
+# message for the meta-repo will be used.
+`;
+
+    return result;
+};
+
+/**
+ * Parse the specified `text` and return an object indicating what (if any)
+ * commit messages to use for the meta-repo and each sub-repo.  The meta-repo
+ * commit message consists of the non-comment lines prior to the first sub-repo
+ * tag.  A sub-repo tag is a line in the form of `# <${sub-repo name}>`.  The
+ * commit message for a sub-repo consists of the non-comment lines between its
+ * tag and the next sub-repo tag, or the end of the file.  If the tag exists
+ * for a sub-repo, but its message is blank, the commit message for the
+ * meta-repo is used. if it has one.  Throw a `UserError` if the same sub-repo
+ * tag is found more than once.
+ *
+ * @param {String} text
+ * @return {Object}
+ * @return {String|null} return.metaMessage commit message for the meta-repo
+ * @return {Object} return.subMessages commit messages for submodules
+ */
+exports.parseSplitCommitMessages = function (text) {
+    assert.isString(text);
+    let metaMessage = null;
+    const seen = new Set();
+    const subMessages = {};
+    const lines = text.split("\n");
+    let start = 0;  // start of current block of lines
+
+    function consumeCurrentBlock(tag, start, end) {
+        if (seen.has(tag)) {
+            throw new UserError(`${tag} was used more than once.`);
+        }
+        seen.add(tag);
+
+        const blockLines = lines.slice(start, end);
+        const message = GitUtil.stripMessageLines(blockLines);
+        if ("*" === tag) {
+            if ("" !== message) {
+                metaMessage = message;
```

Comment:
To be fair, an error message of "* was used more than once" might not be the most clear.